### ISSUE TITLE
RFE: allow setting static dhcp assignments for libvirt VMs

### ISSUE THREAD

**Issue by mcornea** [CentOS-PaaS-SIG/linchpin#961]
Sometimes we need to set static DHCP assignments for the libvirt networks to make sure the VMs obtain consistent ip addresses with the ones set in the inventory. In order to do this we can use virsh net-update after the VMs/networks were created:

```raw
virsh net-update $network_name add ip-dhcp-host "<host mac=$mac_address_of_if_on_network_name name=$vm_name ip=$current_ip_address_assigned_via_dhcp />" --live --config
```

**Comment by abraverm** [comment]
The problem is that node must be (?) deployed after network  and we need to update network setting after deployment.

I see five options here:
1. The simplest will be a post hook that will run a playbook for this. However this is more workaround for something that should be part of deployment process (IMHO).
2. A bit more advanced solution will be adding [DHCP host entries](https://jamielinux.com/docs/libvirt-networking-handbook/appendix/dhcp-host-entries.html) to libvirt network template. This will require setting a static MAC address in node and network setting, which is bothersome to manage.
3. Similar to (2) but a bit different, is to [set IP](https://libvirt.org/formatdomain.html#ipconfig) on node side.
4. Another option will be adding the command to provision_libvirt_node playbook that will update the network without a restarting it. I personally prefer independent configuration, that is not configuring libvirt network from node deployment.
5. Aggregative tasks - where number of provisions are updated after initial deployment. I don't think it will be easy to develop.

**Comment by samvarankashyap** [comment]
@abraverm could you please give an update this issue. 
Thank you :+1:

**Comment by JunqiZhang0** [comment]
@mcornea Do you mean if you run the command you mentioned above could solve the problem?

**Comment by mcornea** [comment]
> @mcornea Do you mean running the command you mentioned above could solve the problem?

Yes, the command solves the problem but it's probably more of a workaround (option #1 that @abraverm mentioned).

**Comment by samvarankashyap** [comment]
fixed by #1198
